docs/src/user-guide/tutorials/auction.md

```diff
@@ -0,0 +1,349 @@
+
+Now you should decide which of these variables will have values that are known at the time the contract is instantiated. It makes sense to set the `beneficiary` right away, so you can add that to the constructor arguments.
+
+The other thing to consider here is *how* the contract will keep track of time. On its own, the contract has no concept of time. However, the contract does have access to the current block timestamp which is measured in seconds since the Unix epoch (Jan 1st 1970). This can be used to measure the time elapsed in a smart contract. In this contract, you can use this concept to set a deadline on the auction. By passing a length of time in seconds to the constructor, you can then add that value to the current block timestamp and create a deadline for bidding to end. Therefore, you should add a `bidding_time` argument to the constructor. Its type can be `u256`.
+
+When you have implemented all this, your contract should look like this:
+
+```rust
+contract Auction {
+    // states
+    auction_end_time: u256
+    beneficiary: address
+    highest_bidder: address
+    highest_bid: u256
+    pending_returns: Map<address, u256>
+    ended: bool
+
+    // constructor
+    pub fn __init__(mut self, ctx: Context, bidding_time: u256, beneficiary_addr: address) {
+        self.beneficiary = beneficiary_addr
+        self.auction_end_time = ctx.block_timestamp() + bidding_time
+    }
+}
+```
+
+Notice that the constructor receives values for `bidding_time` and `beneficiary_addr` and uses them to initialize the contract's `auction_end_time` and `beneficiary` variables.
+
+The other thing to notice about the constructor is that there are two additional arguments passed to the constructor: `must self` and `ctx: Context`.
+
+#### self
+
+`self` is used to represent the specific instance of a Contract. It is used to access variables that are owned by that specific instance. This works the same way for Fe contracts as for, e.g. 'self' in the context of classes in Python, or `this` in Javascript.
+
+Here, you are not only using `self` but you are prepending it with `mut`. `mut` is a keyword inherited from Rust that indicates that the value can be overwritten - i.e. it is "mutable". Variables are not mutable by default - this is a safety feature that helps protect developers from unintended changes during runtime. If you do not make `self` mutable, then you will not be able to update the values it contains.
+
+
+#### Context
+
+Context is used to track deadlines for requests. It gives the contract the ability to cancel functions that have taken too long to run, and provides a structure for storing values that are scoped to a particular request so that they can be accessed anywhere in the call chain. This will be familiar to Go and Rust developers, as Context is often used in the same way in those languages and is frequently employed when working with external API calls. It is conventional to name the context object `ctx`. 
```

Comment:
>Context is used to track deadlines for requests. It gives the contract the ability to cancel functions that have taken too long to run, and provides a structure for storing values that are scoped to a particular request so that they can be accessed anywhere in the call chain.

That part is incorrect. We don't track deadlines for requests or have the ability to cancel functions that take too long (in fact, that doesn't make sense in the context of the EVM as functions can not reach out for things external to the blockchain and everything is ultimately bounded by gas usage).

Context is simply a way for us to make certain access explicit and hence not need extra keywords such as `view` or `pure` like solidity does. Here's a good read on the history of context. I recommend to read it and then re-write this part: https://github.com/ethereum/fe/issues/558

Reply:
ok, thanks, fixing in next commit